### ISSUE TITLE
Malformed URL for transitive dependencies - RC6

### ISSUE THREAD

**Issue by novacekm** [coursier/coursier#596]
I have tried to upgrade drools to version `7.0.0.Final` from `6.5.0.Final`, but I run into malformed URL for transitive dependencies. 

`libraryDependencies += "org.drools" % "drools-compiler" % "7.0.0.Final"` causes the following error:
```
> update
[info] Updating testProject
[info] Resolved testProject dependencies
[trace] Stack trace suppressed: run last *:coursierResolutions for the full output.
[error] (*:coursierResolutions) coursier.ResolutionException: Encountered 3 error(s) in dependency resolution:
[error]   org.kie:kie-bom:${project.version}:
[error]     download error:
[error]       Caught java.io.IOException: Server returned HTTP response code: 400 for URL: https://foo.com/repository/maven-public/org/kie/kie-bom/$%7Bproject.version%7D/kie-bom-$%7Bproject.version%7D.pom (Server returned HTTP response code: 400 for URL: https://foo.com/repository/maven-public/org/kie/kie-bom/$%7Bproject.version%7D/kie-bom-$%7Bproject.version%7D.pom)
[error]       Caught java.io.IOException: Server returned HTTP response code: 400 for URL: https://foo.com/repository/ivy/org.kie/kie-bom/${project.version}/ivys/ivy.xml (Server returned HTTP response code: 400 for URL: https://foo.com/repository/ivy/org.kie/kie-bom/${project.version}/ivys/ivy.xml)
[error]     not found:
[error]       /Users/xyz/.m2/repository/org/kie/kie-bom/$%7Bproject.version%7D/kie-bom-$%7Bproject.version%7D.pom
[error]       /Users/xyz/.ivy2/local/org.kie/kie-bom/${project.version}/ivys/ivy.xml
[error]   org.kie:kie-api::
[error]     not found:
[error]       /Users/xyz/.m2/repository/org/kie/kie-api/kie-api-.pom
[error]       /Users/xyz/.ivy2/local/org.kie/kie-api/ivys/ivy.xml
[error]       https://foo.com/repository/maven-public/org/kie/kie-api//kie-api-.pom
[error]       https:/foo.com/repository/ivy/org.kie/kie-api//ivys/ivy.xml
[error]   org.kie:kie-internal::
[error]     not found:
[error]       /Users/xyz/.m2/repository/org/kie/kie-internal/kie-internal-.pom
[error]       /Users/xyz/.ivy2/local/org.kie/kie-internal/ivys/ivy.xml
[error]       https://foo.com/repository/maven-public/org/kie/kie-internal//kie-internal-.pom
[error]       https://foo.com/repository/ivy/org.kie/kie-internal//ivys/ivy.xml
```
where `foo.com` is our Nexus repository. You can see that the placeholders such as `$%7Bproject.version%7D` are not filled in correctly. However, when using drools version `6.5.0.Final`, there are no problems.

**Comment by nvollmar** [comment]
Looks similiar as I reported back for RC2 https://github.com/coursier/coursier/issues/535

**Comment by novacekm** [comment]
Looks similar, but it is not solved with RC3 and any earlier or later versions :(

**Comment by alexarchambault** [comment]
I have a fix for it locally that I'll push after some clean-up (that roughly allows more intricate things during property substitution...)

**Comment by novacekm** [comment]
Cool. Thanks a lot @alexarchambault

**Comment by alexarchambault** [comment]
This should be fixed since `1.0.0-RC7`. Re-open if you still have issues with that.
